import.sh

```diff
@@ -1,4 +1,32 @@
 #!/usr/bin/env sh
+#
+# This script is meant to be used to update the version of rocksdb in this repo.
+#
+# To use it, update the version number in the URL to the tar.gz file below, then
+# run it. If you're lucky, everything will just work, but you may get an error
+# about a patch not applying cleanly. If that's the case, try to fix up the
+# patch file so that it will apply cleanly, then reset the rest of the
+# repository and try again.
+#
+# Once things have run cleanly, you should have a fairly large diff of the
+# actual rocksdb code, potentially along with some added or removed symlink
+# files in the base of the repository.
+#
+# Once you've successfully run this script, you should also manually verify that
+# the recommended build flags for darwin and linux mostly match up with what's
+# specified in the cgo_flags.go file in the base of the repo.
+# To do so go through the below process on both a mac machine and a linux
+# machine (or a linux docker container):
+#   cd internal
+#   ./build_tools/build_detect_platform tmp.txt
+#   # Manually compare the PLATFORM_CCFLAGS and PLATFORM_CXXFLAGS to what's in
+#   # cgo_flags.go
+#
+# Ask @tamird if you run into issues along the way.
+#
+# After committing locally and before trying to build/test cockroachdb, you
+# should run the command below to ensure your repo is in a clean state:
+#   git clean -dxf
```

Comment:
You would, but that's pretty hard to prevent.

On Thu, Aug 25, 2016 at 3:19 PM, Tobias Schottdorf <notifications@github.com
> wrote:

> In import.sh
> <https://github.com/cockroachdb/c-rocksdb/pull/26#discussion_r76307749>:
>
> > +#
> > +# Once you've successfully run this script, you should also manually verify that
> > +# the recommended build flags for darwin and linux mostly match up with what's
> > +# specified in the cgo_flags.go file in the base of the repo.
> > +# To do so go through the below process on both a mac machine and a linux
> > +# machine (or a linux docker container):
> > +#   cd internal
> > +#   ./build_tools/build_detect_platform tmp.txt
> > +#   # Manually compare the PLATFORM_CCFLAGS and PLATFORM_CXXFLAGS to what's in
> > +#   # cgo_flags.go
> > +#
> > +# Ask @tamird if you run into issues along the way.
> > +#
> > +# After committing locally and before trying to build/test cockroachdb, you
> > +# should run the command below to ensure your repo is in a clean state:
> > +#   git clean -dxf
>
> If I put random files in my c-rocksdb (not that I should, but that
> usually doesn't stop me) would I not accidentally commit them at the end?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/c-rocksdb/pull/26/files/a8e775b9ddd775e84953b8694fa9cfec6e4c5c95#r76307749>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPGJRhmqzpduALTot9-7W2modePs-ks5qjeqqgaJpZM4JtUBm>
> .
>